### PR TITLE
feat(user-agent): Adding User-Agent in headers while making connection to browser

### LINKED ISSUE #6736: [Question] Determine playwright Version
How do I determine the playwright version running on the machine when user makes websocket connection?

### PR DESCRIPTION
**Pull request by karanshah-browserstack**
feat(user-agent): Adding User-Agent in headers while making a connection to the browser.

Adding User-Agent in connection headers while connecting to the browser.
Example: `'Playwright/1.12.0-next (x64/darwin/20.4.0)'`

### CODE REVIEW

File: src/client/browserType.ts [hunk 1]
```diff
@@ -109,12 +109,14 @@ export class BrowserType extends ChannelOwner<channels.BrowserTypeChannel, chann
 
   async connect(params: ConnectOptions): Promise<Browser> {
     const logger = params.logger;
+    const paramsHeaders = params.headers ? params.headers : {};
+    paramsHeaders['User-Agent'] = getUserAgent();
```
Comment (yury-s) [inline src/client/browserType.ts:113]: This is way it cannot be overridden by the client, let's change it to something like `Object.assign({'User-Agent': getUserAgent()}, params.headers)`

File: tests/browsertype-connect.spec.ts [hunk 2]
```diff
@@ -76,13 +76,13 @@ test('should send extra headers with connect request', async ({browserType, star
     browserType.connect({
       wsEndpoint: `ws://localhost:${server.PORT}/ws`,
       headers: {
-        'User-Agent': 'Playwright',
+        'framework': 'Playwright',
```
Comment (yury-s) [inline tests/browsertype-connect.spec.ts:79]: users should be able to override it

**Review by yury-s — CHANGES_REQUESTED** [review]
Please add tests for the new header (see browsertype-connect.spec.ts)

File: tests/browsertype-connect.spec.ts (hunk 2 above)
Comment (karanshah-browserstack) [reply tests/browsertype-connect.spec.ts]: Done

File: src/client/browserType.ts (hunk 1 above)
Comment (karanshah-browserstack) [reply src/client/browserType.ts]: Resolved. Committed the change

**Review by yury-s — APPROVED** [review]
I feel like we may need to add client library language as well. We can do it in the future though as need be.

**Pull request merged**